craft_parts/plugins/rust_plugin.py

```diff
@@ -99,7 +99,7 @@ def validate_environment(
                     plugin_name="rust",
                     part_dependencies=part_dependencies,
                 )
-            options.rust_channel = "none"
+            options.rust_channel = "none"  # type: ignore[misc]
```

Comment:
The type is already optional string, why is this necessary?

Reply:
When I wrote this last week, I thought we had tested the rust plugin to include this, but looking again we haven't.

The complaint was:

```
craft_parts/plugins/rust_plugin.py:103: error: Property "rust_channel" defined in "RustPluginProperties" is read-only  [misc]
```

So this is actually a perfect example of mypy uncovering a bug for us! :tada: 

I've created #559 for that and updated the file with an explanatory comment and a note about removing this ignore comment once we fix it.